compiler/rustc_passes/src/check_attr.rs

```diff
@@ -791,6 +793,26 @@ impl CheckAttrVisitor<'tcx> {
             .emit();
         false
     }
+
+    /// Outputs an error for `#[allow_internal_unstable]` which can only be applied to macros.
+    /// (Allows proc_macro functions)
+    fn check_rustc_allow_const_fn_unstable(
+        &self,
+        attr: &Attribute,
+        span: &Span,
+        target: Target,
+    ) -> bool {
+        if let Target::Fn | Target::Method(_) = target {
+            // FIXME Check that this isn't just a function, but a const fn
```

Comment:
you an use `self.tcx.is_const_fn_raw(self.tcx.hir().local_def_id(hir_id))` to check whether the function should be treated as a const fn.